packages/webdriver-utils/src/metadata/commonDesktopMetaDataResolver.js

```diff
@@ -0,0 +1,43 @@
+export default class CommonDesktopMetaDataResolver {
+  constructor(driver, opts = {}) {
+    this.driver = driver;
+    this.capabilities = opts;
+  }
+
+  browserName() {
+    return this.capabilities.browserName.toLowerCase();
+  }
+
+  osName() {
+    let osName = this.capabilities.osVersion;
+    if (osName) return osName.toLowerCase();
+
+    osName = this.capabilities.platform;
+    return osName;
+  }
+
+  // desktop will show this as browser version
+  osVersion() {
+    return this.capabilities.version.split('.')[0];
+  }
+
+  deviceName() {
+    return this.browserName() + '_' + this.osVersion() + '_' + this.osName();
+  }
+
+  orientation() {
+    return 'landscape';
+  }
+
+  async windowSize() {
```

Comment:
do error handling for any non required functions [ not sure if this is non required or not, its just a general comment ]

Reply:
windowSize is required because we analyze browser window size as our width and height of screenshot.